akka-http-core/src/main/scala/akka/http/javadsl/ConnectionContext.scala

```diff
@@ -5,26 +5,38 @@
 package akka.http.javadsl
 
 import java.util.{ Optional, Collection => JCollection }
+import javax.net.ssl.{ SSLContext, SSLParameters }
+
+import com.typesafe.sslconfig.akka.AkkaSSLConfig
 
+import akka.actor.ClassicActorSystemProvider
 import akka.annotation.DoNotInherit
 import akka.http.scaladsl
 import akka.japi.Util
 import akka.stream.TLSClientAuth
-import com.typesafe.sslconfig.akka.AkkaSSLConfig
-import javax.net.ssl.{ SSLContext, SSLParameters }
 
 import scala.compat.java8.OptionConverters
 
 object ConnectionContext {
   //#https-context-creation
+  def httpsClient(sslContext: SSLContext): HttpsConnectionContext = // ...
```

Comment:
Should we create a variant without parameters that just takes `SSLContext.getDefault()`?

Reply:
Would there be a reason to use it? Wouldn't that be used if you don't configure anything else?

Reply:
Ah, good point - I think the default currently still uses ssl-config.

I agree we should probably switch that to `ConnectionContext.https(Client|Server)(SSLContext.getDefault())` and call that out explicitly in the migration docs.

Or do you think that's a too invasive change in behaviour? I guess we could do a "if there is config, warn and use ssl-config, otherwise use SSLContext.getDefault()"?

Reply:
> Or do you think that's a too invasive change in behaviour? I guess we could do a "if there is config, warn and use ssl-config, otherwise use SSLContext.getDefault()"?

How would we figure out "if there is config"? If that's possible, it would be nice to make that distinction.